fairseq/sequence_generator.py

```diff
@@ -153,6 +163,8 @@ def generate(
         # initialize buffers
         scores = src_tokens.new(bsz * beam_size, max_len + 1).float().fill_(0)
         scores_buf = scores.clone()
+        log_ps_t = src_tokens.data.new(bsz * beam_size, max_len + 1).float().fill_(0)
```

Comment:
yeah, log_ps_t is only used for stochastic beam search, and I set it as None by default, i.e., step(, log_ps_t=None). I thought this for a while, and decided to keep it there since I don't want to add if, else when calling search.step() :-)